subprojects/docs/src/snippets/artifacts/generatedFileDependencies/kotlin/build.gradle.kts

```diff
@@ -17,9 +17,10 @@ tasks.register("compile") {
 }
 
 tasks.register("list") {
-    dependsOn(configurations["compileClasspath"])
+    val compileClasspath: Provider<out FileCollection> = configurations.compileClasspath
```

Comment:
What if `Kotlin DSL` uses the same as Groovy? Does lazy configuration adds anything valuable?

For instance:

```kotlin
val compileClasspath: FileCollection = configurations.compileClasspath.get()
```

Then the samples would be more-or-less the same in both Groovy and Kotlin.

Reply:
Well, I did it the same way it was in the example where we commented recently.

Reply:
I think the documentation should stick to the "same" APIs whenever it is possible.
I believe people often switch to the "Gradle-Kotlin" language to learn Gradle DSL, so it would be super-weird if Groovy DSL used `tasks.create` while Kotlin used `tasks.register` for the same sample.

Here's the other PR: https://github.com/gradle/gradle/pull/21555#discussion_r960958381

Reply:
I tend to agree with @vlsi.

Reply:
Well, me too.
As I said, I just wanted to stay consistent with the example fix.
I changed it now as @bamboo also agrees. :-)